src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java

```diff
@@ -0,0 +1,469 @@
+import de.saxsys.mvvmfx.utils.validation.ValidationStatus;
+import de.saxsys.mvvmfx.utils.validation.Validator;
+
+public class GroupDialogViewModel {
+
+    // Basic Settings
+    private final StringProperty nameProperty = new SimpleStringProperty("");
+    private final StringProperty descriptionProperty = new SimpleStringProperty("");
+    private final StringProperty iconProperty = new SimpleStringProperty("");
+    private final ObjectProperty<Color> colorProperty = new SimpleObjectProperty<>();
+    private final ListProperty<GroupHierarchyType> groupHierarchyListProperty = new SimpleListProperty<>();
+    private final ObjectProperty<GroupHierarchyType> groupHierarchySelectedProperty = new SimpleObjectProperty<>();
+
+    // Type
+    private final BooleanProperty typeExplicitProperty = new SimpleBooleanProperty();
+    private final BooleanProperty typeKeywordsProperty = new SimpleBooleanProperty();
+    private final BooleanProperty typeSearchProperty = new SimpleBooleanProperty();
+    private final BooleanProperty typeAutoProperty = new SimpleBooleanProperty();
+    private final BooleanProperty typeTexProperty = new SimpleBooleanProperty();
+
+    // Option Groups
+    private final StringProperty keywordGroupSearchTermProperty = new SimpleStringProperty("");
+    private final StringProperty keywordGroupSearchFieldProperty = new SimpleStringProperty("");
+    private final BooleanProperty keywordGroupCaseSensitiveProperty = new SimpleBooleanProperty();
+    private final BooleanProperty keywordGroupRegexProperty = new SimpleBooleanProperty();
+
+    private final StringProperty searchGroupSearchTermProperty = new SimpleStringProperty("");
+    private final BooleanProperty searchGroupCaseSensitiveProperty = new SimpleBooleanProperty();
+    private final BooleanProperty searchGroupRegexProperty = new SimpleBooleanProperty();
+
+    private final BooleanProperty autoGroupKeywordsOptionProperty = new SimpleBooleanProperty();
+    private final StringProperty autoGroupKeywordsFieldProperty = new SimpleStringProperty("");
+    private final StringProperty autoGroupKeywordsDelimiterProperty = new SimpleStringProperty("");
+    private final StringProperty autoGroupKeywordsHierarchicalDelimiterProperty = new SimpleStringProperty("");
+    private final BooleanProperty autoGroupPersonsOptionProperty = new SimpleBooleanProperty();
+    private final StringProperty autoGroupPersonsFieldProperty = new SimpleStringProperty("");
+
+    private final StringProperty texGroupFilePathProperty = new SimpleStringProperty("");
+
+    private Validator nameValidator;
+    private Validator nameContainsDelimiterValidator;
+    private Validator sameNameValidator;
+    private Validator keywordRegexValidator;
+    private Validator keywordSearchTermEmptyValidator;
+    private Validator searchRegexValidator;
+    private Validator searchSearchTermEmptyValidator;
+    private CompositeValidator validator = new CompositeValidator();
```

Comment:
I'm not entirely sure if this helps, but maybe worth a try:  In the shared database dialog I initialized this Composite Validator in the ctor and immediately added all validators and that worked without any NPE

       formValidator = new CompositeValidator();
        formValidator.addValidators(databaseValidator, hostValidator, portValidator, userValidator);

Reply:
Tried that.
The validator works fine, it's the lookupbutton method which doesn't work. The buttons seem to be initialized after the rest of the dialog. Nevertheless, it works now. Don't know why it makes a difference between binding and adding a listener, but it works. So... Whatever. ;)